demos/glacial_isostatic_adjustment/gia_adjoint_ice/gia_adjoint_ice.py

```diff
@@ -0,0 +1,717 @@
+#
+# The expression on the left-hand side is termed the second-order Taylor remainder. This term's convergence rate of $O(h^2)$ is a robust indicator for
+# verifying the computational implementation of the gradient calculation. Essentially, if you halve the value of $h$, the magnitude
+# of the second-order Taylor remainder should decrease by a factor of 4.
+#
+# We employ these so-called *Taylor tests* to confirm the accuracy of the determined gradients. The theoretical convergence rate is
+# $O(2.0)$, and achieving this rate indicates that the gradient information is accurate down to floating-point precision.
+#
+# ### Performing Taylor Tests
+#
+# In our implementation, we perform a second-order Taylor remainder test for each term of the objective functional. The test involves
+# computing the functional and the associated gradient when randomly perturbing the initial temperature field, $T_{ic}$, and subsequently
+# halving the perturbations at each level.
+#
+# Here is how you can perform a Taylor test in the code:
+
+h = Function(normalised_ice_thickness)
+h.dat.data[:] = np.random.random(h.dat.data_ro.shape)
+taylor_test(reduced_functional, normalised_ice_thickness, h)
+
+# Now that we have verified our gradient is correct, let's start setting up an inversion. First of all we will define some bounds that we enforce the control to lie within. For this problem the lower bound of zero ice thickness is particularly important, as we do not want negative ice thicknesses!
+
+# +
+ice_thickness_lb = Function(normalised_ice_thickness.function_space(), name="Lower bound ice thickness")
+ice_thickness_ub = Function(normalised_ice_thickness.function_space(), name="Upper bound ice thickness")
+ice_thickness_lb.assign(0.0)
+ice_thickness_ub.assign(5)
+
+bounds = [ice_thickness_lb, ice_thickness_ub]
+# -
+
+# Next we setup a pyadjoint minimization problem. We tweak GADOPT's default minimisation parameters (found in `gadopt/inverse.py`) for our problem. We limit the number of iterations to 20 just so that the demo is quick to run. We also increase the size of the initial radius of the trust region so that the inversion gets going a bit quicker than the default setting.
+
+# +
+minimisation_problem = MinimizationProblem(reduced_functional, bounds=bounds)
+
+minimisation_parameters["Status Test"]["Iteration Limit"] = 20
+minimisation_parameters["Step"]["Trust Region"]["Initial Radius"] = 1e4
+
+optimiser = LinMoreOptimiser(
+    minimisation_problem,
+    minimisation_parameters,
+    checkpoint_dir="optimisation_checkpoint",
+)
+# Restart file for optimisation...
+updated_ice_thickness_file = VTKFile("update_ice_thickness_dispvel1.pvd")
+updated_out_file = VTKFile("updated_out.pvd")
+J_list = []
+# -
+
+
+# Now let's run the inversion!
+
+# +
+optimiser.run()
+
+# If we're performing mulitple successive optimisations, we want
+# to ensure the annotations are switched back on for the next code
+# to use them
+continue_annotation()
+
+# -
```

Comment:
```suggestion
# Now let's run the inversion!

optimiser.run()

# If we're performing mulitple successive optimisations, we want
# to ensure the annotations are switched back on for the next code
# to use them

continue_annotation()
```

Make the continue annotation step a bit more explicit with its own markdown cell